src/ToricVarieties/cohomCalg/ShiftedCones/constructors.jl

```diff
@@ -0,0 +1,66 @@
+#########################
+# 1: The Julia type for shifted cones
+#########################
+
+@attributes mutable struct ShiftedCone
+    gens::fmpz_mat
+    offset::fmpz_mat
+    function ShiftedCone(gens::fmpz_mat, offset::fmpz_mat)
+        if ncols(gens) !== ncols(offset)
+            throw(ArgumentError("The generators must consist of as many integers as the offset."))
+        end
+        return new(gens, offset)
+    end
+end
+export ShiftedCone
+
+@doc Markdown.doc"""
+    ShiftedCone(gens::Vector{Vector{Int64}}, shift::Vector{Int64})
+
+Computes the shifted cones with generators `gens`
+and affine offset `shift`.
+
+# Examples
+A shifted cone `S` is a convex polyhedral cone shifted by an affine offset `a`.
+```jldoctest
+julia> ray_generators = [[1,2,3],[4,5,6]]
+2-element Vector{Vector{Int64}}:
+ [1, 2, 3]
+ [4, 5, 6]
+
+julia> shift = [-3,-2,-1]
+3-element Vector{Int64}:
+ -3
+ -2
+ -1
+
+julia> ShiftedCone(ray_generators, shift)
+A shifted cone in ambient dimension 3
+```
+"""
+ShiftedCone(gens::Vector{Vector{Int64}}, shift::Vector{Int64}) = ShiftedCone(matrix(ZZ, gens), matrix(ZZ, [shift]))
+ShiftedCone(gens::Vector{Vector{fmpz}}, shift::Vector{fmpz}) = ShiftedCone(matrix(ZZ, gens), matrix(ZZ, [shift]))
+
+
+######################
+# 2: Equality
+######################s
+
+function Base.:(==)(sc1::ShiftedCone, sc2::ShiftedCone)
+    if ambient_dim(sc1) != ambient_dim(sc2)
+        return false
+    end
+    if offset(sc1) != offset(sc2)
+        return false
+    end
+    return cone(sc1) == cone(sc2)
```

Comment:
Could also just write
```suggestion
    return ambient_dim(sc1) == ambient_dim(sc2) && offset(sc1) == offset(sc2) && cone(sc1) == cone(sc2)
```

Reply:
Thank you!